python/torch_mlir_e2e_test/test_suite/reduction.py

```diff
@@ -49,6 +49,25 @@ def ReduceSumDtypeFloatModule_basic(module, tu: TestUtils):
 
 # ==============================================================================
 
+class ReduceSumElementTypeBoolModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args([
+        None,
+        ([-1, -1, -1], torch.bool, True),
+    ])
+    def forward(self, a):
+        return torch.sum(a).to(torch.bool)
+
+
+@register_test_case(module_factory=lambda: ReduceSumElementTypeBoolModule())
+def ReduceSumElementTypeBoolModule_basic(module, tu: TestUtils):
+    module.forward(tu.rand(3, 4, 5).to(torch.bool))
```

Comment:
When converting floats to bool, PyTorch only assigns `False` to the value `0.0`, so `tu.rand` is almost guaranteed to only produce `True` values.

To generate a bool tensor, you should do `tu.randint(3, 4, 5, high=2).to(torch.bool)`